students/Sean_Tasaki/Lesson06/mailroom_part4.py

```diff
@@ -0,0 +1,106 @@
+"""
+Sean Tasaki
+5/21/2018
+Lesson06.mailroom_part_4
+"""
+from collections import defaultdict
+
+donor_dict = defaultdict(lambda : [0, 0, 0], {'Bob Dylan': [20000.00, 3, 40000.00], 'Leonard Cohen' : [600.00, 2, 300.00], 'Soren Kierkegaard' : [1000.00, 2, 500.00], 'Italo Calvino' : [20000.00, 4, 5000.00], 'Florence Feist' : [300.00, 3, 100.00]})
+
+
+def main_menu():
+    main_menu_dict = {'1': thank_you, '2': create_report, '3': create_thank_you_letters, '4': quit}
+    main_prompt = "Enter 1-4 from the following options: (1) Send a Thank You to a Donor, (2) Create a Report, (3) Write a Thank You Letter to All Donors (4) Quit\n >> "  
+    main_menu_response(main_prompt, main_menu_dict)
+
+def main_menu_response(prompt, main_menu_dict):
+    try:
+        while True:
+            response = input(prompt)
+            main_menu_dict.get(response)()
+            if main_menu_dict[response]() == "exit menu":
+                exit()
+    except TypeError:
+        print("Enter a number between 1-4.")    
+        return main_menu()
+
+def thank_you():
+    name = input("Enter the first and last name of the donor or enter 'list' to see a list of previous donor names or enter Q to exit to main menu\n> ")
+    if name.lower() == 'list':
+        for donor in donor_dict:
+            print(donor)
+        thank_you()
+    elif name.upper() == 'Q':
+        main_menu()
+    else:
+        add_donor_to_dict(name.title())
+    try:
+        if donor_dict[name.title()][1] == 0:
+            amount = input(f"{name.title()} is a new donor. Enter the donation amount.\n>> ")
+            add_donor_amount(name.title(), amount)
+            print(f"Thank you {name.title()} for becoming a new donor to our charity! Your genereous donation of ${float(amount):.2f} is much appreciated.")
+        else:
+            amount = input(f"{name.title()} is a previous donor. Enter the donation amount.\n>> ")
+            add_donor_amount(name.title(), amount)
+            print(f"Thank you {name.title()} for your loyal support to our charity! Your genereous donation of ${float(amount):.2f} is much appreciated.")
+
```

Comment:
You want to include as little code in your `try` block as possible, and only the code that you expect to raise the error that you are handling in your `except` block